### PR TITLE
Add "make_inductor" (for directors, aka inductors on "the door")

### PR DESCRIPTION
**Pull request by castaway**
/make_inductor &lt;name&gt; on &lt;tool&gt;

Also made "/inductions" default to current user if not supplied

### CODE REVIEW

**Review by theorbtwo — APPROVED** [review]
Not looked deeply, but at first glance seems good.

File: lib/AccessSystem/TelegramBot.pm [hunk 1]
```diff
@@ -499,6 +497,101 @@ sub inductions ($self, $message) {
...
+            # member is not a director, and not an inductor on the tool
+            return $message->$reply("You're not allowed to do that");
+        }
+        my $inducted = $person->find_or_create_related('allowed', { tool_id => $tool->id });
+        $inducted->update({ is_admin => 1 });
+        return $message->$reply("Ok, made " . $person->name ." an inductor on " . $tool->name);
+    }
+
+    # we didn't find an exact tool, display buttons to pick from instead
+    # (the buttons callback will re-run this whole method with $args set)
```
Comment (lucas-wilkins) [inline lib/AccessSystem/TelegramBot.pm:575]: :)

File: lib/AccessSystem/TelegramBot.pm [hunk 2]
```diff
@@ -499,6 +497,101 @@ sub inductions ($self, $message) {
...
+    return if !$door;
+
+    ## Must be an admin, or existing inductor
+    my $allowed = $member->allowed->find({ tool_id => $door->id });
+    return if !$allowed;
+    # if (!$allowed->is_admin) {
+    #     return $message->$reply("You're not allowed to do that");
+    # }
+
+    if (!$args && $message->text =~ m{^/make_inductor\s([\w\s]+)\son\s([\w\s\d]+)$}) {
```
Comment (lucas-wilkins) [inline lib/AccessSystem/TelegramBot.pm:525]: Will \w capture names that include accents?

Comment (lucas-wilkins) [reply lib/AccessSystem/TelegramBot.pm:525]: Actually a general thing I've been wondering about re this system

**Review by lucas-wilkins — APPROVED** [review]
Just one question

File: lib/AccessSystem/TelegramBot.pm (hunk 2 above)
Comment (castaway) [reply lib/AccessSystem/TelegramBot.pm:525]: Good question it should ... just tested with some random polish string : "/inductions małego psa" (thats an L with a strike through in the middle) and it does look that up just fine.. however it won't pick up double-barrelled (aka "with -") surnames, or irish ones with ' in.. so should probably expand it a bit.. 

Sadly regexp::common (a library of common regexes) doesn't seem to have one for "names"..

**Pull request merged**